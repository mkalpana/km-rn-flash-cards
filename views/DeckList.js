import React, { Component } from 'react';
import { View, Text, StyleSheet, FlatList, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { grey } from '../utils/colors';
import { fetchDecks } from '../actions';
import { connect } from 'react-redux';

export class DeckList extends Component{
  componentDidMount() {
    this.props.fetchDecks();
  }

  renderCard = (deck) => {
    const { navigation } = this.props;
    return (deck && deck.title && deck.questions &&
      <TouchableOpacity onPress={() => navigation.navigate('Deck', { title: deck.title })}>
        <View key={deck.title} style={styles.card}>
          <Text style={styles.deckTitle}>{deck.title}</Text>
          <Text style={styles.cardCount}>{deck.questions && deck.questions.length} cards</Text>
        </View>
      </TouchableOpacity>
    );
  };

  render() {
    const { decks } = this.props;
    return (
      <View style={styles.container}>
        <FlatList
          data={decks}
          renderItem={({ item }) => this.renderCard(item)}
          keyExtractor={(item, index) => index}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  card: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 50,
    borderBottomWidth: 1,
    borderBottomColor: 'grey'
  },
  deckTitle: {
    color: 'black',
    fontSize: 40,
  },
  cardCount: {
    color: 'grey',
    fontSize: 16,
  }
});

DeckList.propTypes = {
  decks: PropTypes.array.isRequired,
  fetchDecks: PropTypes.func.isRequired,
};

DeckList.defaultPropTypes = {
  decks: []
};

const mapStateToProps = (state) => {
  const decksList = state.decks;
  const decks = Object.keys(decksList).map(title => decksList[title]);
  return { decks };
};

export default connect(mapStateToProps, {
  fetchDecks
})(DeckList);

