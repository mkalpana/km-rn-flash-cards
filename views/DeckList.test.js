import React from 'react';
import { DeckList } from './DeckList';
import renderer from 'react-test-renderer';

const decks = {
  React: { title: 'React', questions: [] },
  JavaScript: { title: 'JavaScript', questions: [] },
};

describe('DeckList', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<DeckList
      decks={Object.keys(decks).map(key => decks[key])}
      fetchDecks={() => decks}
    />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
