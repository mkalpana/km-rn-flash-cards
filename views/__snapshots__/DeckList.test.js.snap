// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckList renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "questions": Array [],
          "title": "React",
        },
        Object {
          "questions": Array [],
          "title": "JavaScript",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={undefined}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "grey",
                "borderBottomWidth": 1,
                "justifyContent": "center",
                "padding": 50,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "fontSize": 40,
                }
              }
            >
              React
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "grey",
                  "fontSize": 16,
                }
              }
            >
              0
               cards
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "grey",
                "borderBottomWidth": 1,
                "justifyContent": "center",
                "padding": 50,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "fontSize": 40,
                }
              }
            >
              JavaScript
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "grey",
                  "fontSize": 16,
                }
              }
            >
              0
               cards
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
